packages/core/saved-objects/core-saved-objects-migration-server-internal/src/model/model.test.ts

```diff
@@ -2599,5 +2602,50 @@ describe('migrations v2 model', () => {
         expect(newState.retryDelay).toEqual(0);
       });
     });
+    describe('skipping reindex if not necessary', () => {
+      const unchangedMappingsState: State = {
+        ...baseState,
+        controlState: 'INIT',
+        kibanaVersion: '7.12.0', // new version!
+        currentAlias: '.kibana',
+        versionAlias: '.kibana_7.12.0',
+        versionIndex: '.kibana_7.11.0_001',
+      };
+      it('INIT -> PREPARE_COMPATIBLE_MIGRATION when desired mappings match current mappings', () => {
+        const res: ResponseType<'INIT'> = Either.right({
+          '.kibana_7.11.0_001': {
+            aliases: {
+              '.kibana': {},
+              '.kibana_7.11.0': {},
+            },
+            mappings: indexMapping,
+            settings: {},
+          },
+        });
+        const newState = model(unchangedMappingsState, res) as PrepareCompatibleMigration;
+
+        expect(newState.controlState).toEqual('PREPARE_COMPATIBLE_MIGRATION');
```

Comment:
ideally we'd test all state that could impact further decision making. It was an oversight to not test for `targetIndexRawMappings` in the other `INIT -> *` transitions where this has an impact. We could address that separately but for this PR testing that `targetIndexRawMappings` is set before the step transition is quite important.